packages/endpoints-organizations-registrar/src/entities/signatories/repos/signatory-status-state-repo-extension.js

```diff
@@ -43,28 +44,68 @@ const signatoryStatusStateRepoExtension = (parent) => ({
         $set: parent.prepModification($set),
       };
     }
+    const filter = isPlainObject(_idOrFilter)
+      ? _idOrFilter
+      : { _id: new ObjectId(_idOrFilter) };
+
     const result = await parent
       .collection()
-      .findOneAndUpdate(
-        parent.prepFilter({ organizationDid: did }),
-        updateStatement,
-        {
-          returnDocument: 'after',
-          includeResultMetadata: true,
-          projection,
-        }
-      );
+      .findOneAndUpdate(parent.prepFilter(filter), updateStatement, {
+        returnDocument: 'after',
+        includeResultMetadata: true,
+        projection,
+      });
     return result.value;
   },
+  findByEvent: async (eventState, eventTimestamp) => {
+    const aggregationPipeline = [
+      {
+        $match: {
+          'events.state': { $nin: ['APPROVED', 'REJECTED'] },
```

Comment:
wdyt about adding a `COMPLETED` state after `APPROVED` or `REJECTED`?

Reply:
@nassan I think introducing completed is more consistent with other implementations of this pattern

Reply:
done, see [b956133](https://github.com/velocitycareerlabs/velocitycore/pull/34/commits/b95613348d7398ed5ab0fec94b755ac759c4eb21)